### PR TITLE
[4.0] Handle tags on copy via batch

### PR DESCRIPTION
**Pull request by bembelimen**
Pull Request for Issue #31127 .

### Summary of Changes
Copy also the tags when items are copied.

### Testing Instructions
See: #31127

### Actual result BEFORE applying this Pull Request
Tags are not copied

### Expected result AFTER applying this Pull Request
Tags are copied

### CODE REVIEW

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 18a8f4ed51c4268dcd3b50b90c49d1b5dbea103a<br /><br />Tested successfully in Beta6-dev of 28 December.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31793">issues.joomla.org/tracker/joomla-cms/31793</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 18a8f4ed51c4268dcd3b50b90c49d1b5dbea103a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31793">issues.joomla.org/tracker/joomla-cms/31793</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31793">issues.joomla.org/tracker/joomla-cms/31793</a>.</sub>

**Comment by ghost** [comment]
add PR-4.0-dev label again.

**Comment by infograf768** [comment]
@SharkyKZ 
Please explain your comment. Thanks.

File: libraries/src/MVC/Model/AdminModel.php [hunk 1]
```diff
@@ -486,6 +486,12 @@ protected function batchCopy($value, $pks, $contexts)
 			// New category ID
 			$this->table->catid = $categoryId;
 
+			if (property_exists($this->table, 'typeAlias') && property_exists($this->table, 'tagsHelper'))
+			{
+				$this->table->newTags = $this->table->tagsHelper->tags;
+
```
Comment (infograf768) [inline libraries/src/MVC/Model/AdminModel.php]: Drone fails because of this blank line

Comment (bembelimen) [reply libraries/src/MVC/Model/AdminModel.php]: Thx

**Comment by infograf768** [comment]
This only works when copying the article.
If it is a simple change (category, language), then we lose the Tag.

**Comment by bembelimen** [comment]
We lose everytime the tags when ```$table->store()``` is called. That is a fundamental problem, how the taggable plugin is implemented. So we have to touch everywhere, where table is stored in the Joomla! environment. (or rewrite the taggable behaviour). So this PR only solves the issue itself.

**Comment by richard67** [comment]
Last change after tests and RTC was only code style, so the tests and the RTC are still valid. Restoring the test result in the tracker in a minute.

**Comment by wilsonge** [comment]
Holding on merging this until I understand whether this behaviour is needed if the underlying issue in the taggable plugin is fixed. I suspect this is independent but want to do a bit of digging

**Comment by wilsonge** [comment]
Done a larger rework here https://github.com/joomla/joomla-cms/pull/31805 that should give us better functionality across all of batch

**Pull request closed without merging**